src/learning/toolkit/net_layer.rs

```diff
@@ -34,20 +34,37 @@ pub trait NetLayer : Debug {
 
 /// Linear network layer
 ///
-/// The parameters are a matrix of weights of size O x I
+/// Represents a fully connected layer with optional bias term
+///
+/// The parameters are a matrix of weights of size I x O
 /// where O is the dimensionality of the output and I the dimensionality of the input
 #[derive(Debug, Clone, Copy)]
 pub struct Linear {
 	/// The number of dimensions of the input
 	input_size: usize,
 	/// The number of dimensions of the output
 	output_size: usize,
+	/// Whether or not to include a bias term
+	has_bias: bool,
 }
 
 impl Linear {
-	/// Construct a Linear layer
-	pub fn new(input_size: usize, output_size: usize) -> Linear {
-		Linear{input_size: input_size, output_size: output_size}
+	/// Construct a new Linear layer
+	pub fn new(input_size: usize, output_size: usize, has_bias: bool) -> Linear {
+		Linear {
+			input_size: input_size + has_bias as usize, 
+			output_size: output_size,
+			has_bias: has_bias
+		}
+	}
+
+	/// Construct a Linear layer with a bias term
+	pub fn default(input_size: usize, output_size: usize) -> Linear {
```

Comment:
I can make that change. `new` will include a bias and there'll be a separate `without_bias`.